tidy3d/web/api/mode.py

```diff
@@ -137,6 +145,106 @@ def run(
+    verbose : bool
+        If True, displays a progress bar. If False, runs silently.
+    max_workers : int
+        Maximum number of concurrent workers to use for processing the batch of simulations.
+    max_retries : int
+        Maximum number of retries for each simulation in case of failure before giving up.
+    retry_delay : int
+        Delay in seconds between retries when a simulation fails.
+    progress_callback_upload : Callable[[float], None], optional
+        Optional callback function called when uploading file with ``bytes_in_chunk`` as argument.
+    progress_callback_download : Callable[[float], None], optional
+        Optional callback function called when downloading file with ``bytes_in_chunk`` as argument.
+
+
+    Returns
+    -------
+    List[ModeSolverData]
+        A list of ModeSolverData objects containing the results from each simulation in the batch. ``None`` is placed in the list for simulations that fail after all retries.
+    """
+    console = get_logging_console()
+
+    # Check type to make sure the user is submitting a list of mode solvers
+    if not all(isinstance(x, ModeSolver) for x in mode_solvers):
+        log.error("[red]Stopped running a batch of mode solver simulations. Please provide a list of mode solvers of type [cyan]`td.plugins.mode.mode_solver.ModeSolver` [red]to run a batch of mode solver simulations.")
+        return
+    
+    num_mode_solvers = len(mode_solvers)
+
+    # Result files where the data is stored after downloading
+    if results_files is None:
+        results_files = [f"{folder_name}/mode_solver_results_{i}.hdf5" for i in range(num_mode_solvers)]
+
+    results = [None] * num_mode_solvers 
+    
+    # Handling simulation to make sure there are more than one tries if simulation produces some error
+    def handle_mode_solver(index, retries=0):
+        try:
+            # We'll create a separate progress bar for batch simulations so we'll set verbose=False here for individual runs
+            result = run(
+                        mode_solver = mode_solvers[index], 
+                         task_name = f"{task_name}_{index}", 
+                         mode_solver_name = f"mode_solver_batch_{index}", 
+                         folder_name = folder_name, 
+                         results_file = results_files[index], 
+                         verbose = False, 
+                         progress_callback_upload = progress_callback_upload, 
+                         progress_callback_download = progress_callback_download
+                         )
+            results[index] = result
+        except Exception as exc:
+            if retries < max_retries:
+                # Wait before retrying
+                time.sleep(retry_delay)  
+                handle_mode_solver(index, retries + 1)
+            else:
+                log.warning(f"The {index}-th mode solver failed after {max_retries} tries.")          
```

Comment:
what are we going to do about it?  Maybe add another sentence that the data returned by this mode solver will be `None`?